src/LiveComponent/src/ComponentWithFormTrait.php

```diff
@@ -179,41 +170,27 @@ private function getFormInstance(): FormInterface
         return $this->formInstance;
     }
 
-    /**
-     * Automatically adds data-model="*" to the form element.
-     *
-     * This makes it so that all fields will automatically become
-     * "models", using their "name" attribute.
-     *
-     * This is for convenience: it prevents you from needing to
-     * manually add data-model="" to every field. Effectively,
-     * having name="foo" becomes the equivalent to data-model="foo".
-     *
-     * To disable or change this behavior, override the
-     * the getDataModelValue() method.
-     */
-    private function useNameAttributesAsModelName(): void
+    private function configureFormView(FormView $form): FormView
     {
-        $modelValue = $this->getDataModelValue();
-        $attributes = $this->getForm()->vars['attr'] ?: [];
-        if (null === $modelValue) {
-            unset($attributes['data-model']);
-        } else {
-            $attributes['data-model'] = $modelValue;
-        }
+        $behavior = new ModelBehavior();
+        $this->configureModelBehavior($behavior);
+        $this->applyDataModelsToForm($form, $behavior);
 
-        $this->getForm()->vars['attr'] = $attributes;
+        return $form;
     }
 
-    /**
-     * Controls the data-model="" value that will be rendered on the <form> tag.
-     *
-     * This default value will cause the component to re-render each time
-     * a field "changes". Override this in your controller to change the behavior.
-     */
-    private function getDataModelValue(): ?string
+    private function applyDataModelsToForm(FormView $form, ModelBehavior $behavior): void
     {
-        return 'on(change)|*';
+        foreach ($form->children as $child) {
+            $attr = $child->vars['attr'] ?? [];
+            $fullName = $child->vars['full_name'] ?? '';
+
+            $attr['data-model'] = $behavior->getModelForField($fullName);
+            $attr['data-successful'] = json_encode($this->wasSubmitted && !count($child->vars['errors'] ?? []));
```

Comment:
This allows us to better determine wether an element has been succesfully validated or not. There is no way to tell from the FormView, since all data is submitted, but errors could be cleared.